homeassistant/components/evohome/water_heater.py

```diff
@@ -0,0 +1,107 @@
+"""Support for WaterHeater devices of (EMEA/EU) Honeywell evohome systems."""
+from datetime import datetime, timedelta
+import logging
+
+import requests.exceptions
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    WaterHeaterDevice)
+from homeassistant.const import (
+    STATE_OFF, STATE_ON)
+
+from . import (
+    EvoDevice,
+    CONF_LOCATION_IDX)
+from .climate import (
+    EVO_FOLLOW, EVO_TEMPOVER, EVO_PERMOVER)
+from .const import (
+    DATA_EVOHOME, GWS, TCS)
+
+_LOGGER = logging.getLogger(__name__)
+
+EVO_STATE_TO_HA = {'On': STATE_ON, 'Off': STATE_OFF}
+HA_STATE_TO_EVO = {v: k for k, v in EVO_STATE_TO_HA.items()}
+
+HA_OPMODE_TO_EVO = {STATE_ON: EVO_FOLLOW, STATE_OFF: EVO_PERMOVER}
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Create the DHW controller."""
+    evo_data = hass.data[DATA_EVOHOME]
+
+    client = evo_data['client']
+    loc_idx = evo_data['params'][CONF_LOCATION_IDX]
+
+    # evohomeclient has exposed no means of accessing non-default location
+    # (i.e. loc_idx > 0) other than using a protected member, such as below
+    tcs_obj_ref = client.locations[loc_idx]._gateways[0]._control_systems[0]  # noqa: E501; pylint: disable=protected-access
+
+    _LOGGER.info(
+        "setup(): Found DHW device, id: %s [%s]",
```

Comment:
Don't include a function name in the log message.

Reply:
have removed